amxmodx/file.cpp

```diff
@@ -744,10 +744,16 @@ enum FileType
 
 struct DirectoryHandle
 {
-	DirectoryHandle(void* handle_, bool valvefs_) : handle(handle_), valvefs(valvefs_) {}
+	DirectoryHandle(CDirectory *handle_) : handle(handle_), valvefs(false) {}
+	DirectoryHandle(FileFindHandle_t handle_) : handle_vfs(handle_), valvefs(true) {}
 
-	void* handle;
-	bool  valvefs;
+	union
+	{
+		CDirectory*      handle;
+		FileFindHandle_t handle_vfs;
+	};
```

Comment:
There is no variant in amtl?

Reply:
Looks like not. Well, C++17 features are not yet a thing in AMTL. Feel free to contribute!